src/auth/auth.service.ts

```diff
@@ -37,4 +38,34 @@ export class AuthService {
       },
     });
   }
+
+  async signIn(signInDto: SignInDto): Promise<UserWithoutPassword> {
+    const { username, password } = signInDto;
+    const user = await this.prismaService.user.findFirst({
+      where: {
+        OR: [
+          {
+            studentCode: username
+          },
+          {
+            email: username
+          }
+        ]
+      },
+    });
+    
+    if (!user) {
+      throw new BadRequestException("User not found");
+    }
+
+    const passwordValid = await bcrypt.compare(password, user.password);
+    if (!passwordValid) {
+      throw new BadRequestException("Invalid password");
+    }
+
+    const { id, email, name, studentCode, avatar, generation, createdAt, updatedAt } = user;
+    const userData = { id, email, name, studentCode, avatar, generation, createdAt, updatedAt };
+    
+    return userData;
```

Comment:
em thử xem nhé, dùng destructuring kết hợp với rest cho hợp lý nhé, chứ destructure nhiều thuộc tính như này không ổn lắm 👍 
```suggestion
    const { password, ...userWithoutPassword } = user;
    return userWithoutPassword;
```